Gems/SceneProcessing/Code/Tests/MeshBuilder/MeshOptimizerComponentTests.cpp

```diff
@@ -0,0 +1,221 @@
+    {
+    public:
+        void SetUp() override
+        {
+            SceneProcessing::InitSceneAPIFixture::SetUp();
+
+            m_systemEntity = m_app.Create({}, {});
+            m_systemEntity->AddComponent(aznew AZ::MemoryComponent());
+            m_systemEntity->AddComponent(aznew AZ::JobManagerComponent());
+            m_systemEntity->Init();
+            m_systemEntity->Activate();
+        }
+        void TearDown() override
+        {
+            m_systemEntity->Deactivate();
+            SceneProcessing::InitSceneAPIFixture::TearDown();
+        }
+
+        static AZStd::unique_ptr<AZ::SceneAPI::DataTypes::IMeshData> MakePlaneMesh()
+        {
+            // Create a simple plane with 2 triangles, 6 total vertices, 2 shared vertices
+            // 0 --- 1
+            // |   / |
+            // |  /  |
+            // | /   |
+            // 2 --- 3
+            const AZStd::array planeVertexPositions = {
+                AZ::Vector3{0.0f, 0.0f, 0.0f},
+                AZ::Vector3{0.0f, 0.0f, 1.0f},
+                AZ::Vector3{1.0f, 0.0f, 1.0f},
+                AZ::Vector3{1.0f, 0.0f, 1.0f},
+                AZ::Vector3{1.0f, 0.0f, 0.0f},
+                AZ::Vector3{0.0f, 0.0f, 0.0f},
+            };
+
+            auto mesh = AZStd::make_unique<AZ::SceneData::GraphData::MeshData>();
+
+            int i = 0;
+            for (const AZ::Vector3& position : planeVertexPositions)
+            {
+                mesh->AddPosition(position);
+                mesh->AddNormal(AZ::Vector3::CreateAxisY());
+
+                // This assumes that the data coming from the import process gives a unique control point
+                // index to every vertex. This follows the behavior of the AssImp library.
+                mesh->SetVertexIndexToControlPointIndexMap(i, i);
+                ++i;
+            }
+
+            mesh->AddFace({0, 1, 2}, 0);
+            mesh->AddFace({3, 4, 5}, 0);
+
+            return mesh;
+        }
+
+        static AZStd::unique_ptr<AZ::SceneData::GraphData::SkinWeightData> MakeSkinData()
+        {
+            auto skinWeights = AZStd::make_unique<AZ::SceneData::GraphData::SkinWeightData>();
+
+            skinWeights->ResizeContainerSpace(6);
+
+            // Add bones 0 and 1 to the skin weights
+            skinWeights->GetBoneId("0");
+            skinWeights->GetBoneId("1");
+
+            skinWeights->AppendLink(0, {/*.boneId=*/0, /*.weight=*/1});
+            skinWeights->AppendLink(1, {/*.boneId=*/0, /*.weight=*/1});
+            skinWeights->AppendLink(2, {/*.boneId=*/0, /*.weight=*/1});
+            skinWeights->AppendLink(3, {/*.boneId=*/1, /*.weight=*/1});
+            skinWeights->AppendLink(4, {/*.boneId=*/1, /*.weight=*/1});
+            skinWeights->AppendLink(5, {/*.boneId=*/1, /*.weight=*/1});
+
+            return skinWeights;
+        }
+
+    private:
+        AZ::ComponentApplication m_app;
+        AZ::Entity* m_systemEntity;
+    };
+
```

Comment:
Minor nitpick: It would be useful to have some simple baseline tests. Things like:

- Make sure the blendshape flag is handled correctly and it doesn't weld when the shape is marked as a blendshape.
- Invalid / null data
- All vertices are duplicates of each other
- No duplicates, nothing to weld